GVRf/Sample/gamepad/src/org/gearvrf/gamepad/SampleViewManager.java

```diff
@@ -72,6 +76,80 @@ public void onInit(GVRContext gvrContext) throws IOException {
     public void onStep() {
     }
 
+    // getCenteredAxis() and some other code are referenced from:
+    // https://developer.android.com/training/game-controllers/controller-input.html
+    private static float getCenteredAxis(MotionEvent event, InputDevice device,
+            int axis, int historyPos) {
+        final InputDevice.MotionRange range = device.getMotionRange(axis,
+                event.getSource());
+
+        // A joystick at rest does not always report an absolute position of
+        // (0,0). Use the getFlat() method to determine the range of values
+        // bounding the joystick axis center.
+        if (range != null) {
+            final float flat = range.getFlat();
+            final float value = historyPos < 0 ? event.getAxisValue(axis)
+                    : event.getHistoricalAxisValue(axis, historyPos);
+
+            // Ignore axis values that are within the 'flat' region of the
+            // joystick axis center.
+            if (Math.abs(value) > flat) {
+                return value;
+            }
+        }
+        return 0;
+    }
+
+    public boolean processJoystickInput(MotionEvent event, int historyPos) {
+        if (mScene.getSceneObjects().isEmpty()) {
+            return false;
+        }
+
+        InputDevice mInputDevice = event.getDevice();
+
+        // Calculate the horizontal distance to move by
+        // using the input value from one of these physical controls:
+        // the left control stick, hat axis, or the right control stick.
+        float x = getCenteredAxis(event, mInputDevice, MotionEvent.AXIS_X,
+                historyPos);
+        float hatx = getCenteredAxis(event, mInputDevice,
+                MotionEvent.AXIS_HAT_X, historyPos);
+        
+        // Google refers to this axis as z, Samsung Gamepad uses RX.
+        float z = getCenteredAxis(event, mInputDevice, MotionEvent.AXIS_RX,
+                historyPos);
+
```

Comment:
When you write "Google refers to..." do you refer to the image in https://developer.android.com/training/game-controllers/controller-input.html?
I find it confusing to print to the log z, rz instead of axis_rx, axis_ry.